src/torchjd/autojac/multi_task_backward.py

```diff
@@ -0,0 +1,176 @@
+from typing import Iterable, Sequence
+
+from torch import Tensor
+
+from torchjd.aggregation import Aggregator
+
+from ._transform import (
+    EmptyTensorDict,
+    Grad,
+    Gradients,
+    Init,
+    Jac,
+    Stack,
+    Store,
+    Subset,
+    Transform,
+    make_aggregation,
+)
+from ._transform.strategy import UnifyingStrategy
+from ._utils import (
+    _as_tensor_list,
+    _check_optional_positive_chunk_size,
+    _check_retain_graph_compatible_with_chunk_size,
+)
+
+
+def multi_task_backward(
+    tasks_losses: Sequence[Tensor],
+    shared_parameters: Iterable[Tensor],
+    shared_representations: Sequence[Tensor] | Tensor,
+    tasks_parameters: Sequence[Iterable[Tensor]],
+    A: Aggregator,
+    retain_graph: bool = False,
+    parallel_chunk_size: int | None = None,
+) -> None:
+    r"""
+    Used for Multi-task learning, this computes the gradient of the ``tasks_losses`` with respect to
+    the ``tasks_parameters`` and stores them in their ``.grad`` fields. It then computes the
+    Jacobian of the ``tasks_losses`` with respect to the ``shared_parameters`` and stores its
+    aggregation by ``A`` into their ``.grad`` fields. The ``tasks_losses`` should be computed from
+    ``shared_activation`` and ``tasks_parameters``.
+
+
+    .. admonition::
+        Example
+
+        The following code snippet showcases a simple usage of ``multi_task_backward``.
+
+            >>> import torch
+            >>>
+            >>> from torchjd import multi_task_backward
+            >>> from torchjd.aggregation import UPGrad
+            >>>
+            >>> p0 = torch.tensor([1.0, 2.0], requires_grad=True)
+            >>> p1 = torch.tensor([1.0, 2.0], requires_grad=True)
+            >>> p2 = torch.tensor([3.0, 4.0], requires_grad=True)
+            >>> shared_parameters = [p0]
+            >>> tasks_parameters = [[p1], [p2]]
+            >>>
+            >>> # Compute arbitrary quantities that are function of the parameters
+            >>> r1 = torch.tensor([-1.0, 1.0]) @ p0
+            >>> r2 = (p0**2).sum() + p0.norm()
+            >>> shared_representations = [r1, r2]
```

Comment:
I think the basic example should have a single shared representation. Having more than 1 is very rare, so it's quite confusing in an example.

Reply:
Since we take `Sequence[Tensor] | Tensor`, this could be directly changed to
```python
# Compute an arbitrary representation that is function of the shared parameters
shared_representation = torch.tensor([-1.0, 1.0]) @ p0
```
directly

Reply:
Actually I would prefer the shared_representation to be a vector (it's more usual)

Reply:
But not a list of tensors imo